spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -33,14 +34,20 @@
 class SpyderIPy3Lexer(IPython3Lexer):
     # Detect !cmd command and highlight them
     tokens = IPython3Lexer.tokens
-    tokens['root'].insert(
-        0, (r'(!)(\w+)(.*\n)', bygroups(Operator, Keyword, Text)))
+    spyder_tokens = [
+        (r'(!)(\w+)(.*\n)', bygroups(Operator, Keyword, using(Python3Lexer))),
+        (r'(%)(\w+)(.*\n)', bygroups(Operator, Keyword, using(Python3Lexer))),
+        ]
+    tokens['root'] = spyder_tokens + tokens['root']
 
 
 class SpyderIPy2Lexer(IPythonLexer):
```

Comment:
Should we keep this class? I think it's not necessary anymore because in Spyder 6 it won't be possible to run Python 2 anymore.

Reply:
Ok if there are no python2 kernels I will remove this